client/src/views/components/PayMethods/PayMethods.ts

```diff
@@ -3,38 +3,72 @@ import handleEvent from '@src/utils/handleEvent';
 import { CheckButton, ETC, Xbox } from '@src/static/image-urls';
 import './PayMethods.scss';
 import { $, removeActiveAttributeOnClass } from '@src/utils/helper';
+import ConfirmWindow from '../Confirm/Confirm';
 
 export default class PayMethod {
   state: any;
   PayWrapper: HTMLElement;
   mode: string = 'Account';
   currentMode: HTMLElement;
   currentCardName: string = '';
+  filter: any;
 
-  constructor({ parent, state }) {
+  constructor({ parent, state, filter }) {
     if (parent === $('.history-form__pay-method')) this.mode = 'historyModal';
 
     this.PayWrapper = document.createElement('div');
 
     parent.appendChild(this.PayWrapper);
-    this.setState(state);
+    this.setProperty(state, filter);
     this.render();
 
     this.currentMode = document.querySelector('#' + this.mode);
     this.currentMode.addEventListener('click', this.onClickHandler.bind(this));
   }
 
+  setProperty(state, filter): void {
+    this.state = state;
+    this.filter = filter;
+  }
+
+  render(): void {
+    this.PayWrapper.innerHTML = `
+        ${this.createPayMethod()}
+        `;
+  }
+
   onClickHandler(e: MouseEvent) {
     this.onClickAddButton(e);
     this.onClickCardChoice(e);
-    // this.onClickXbox(e);
+    this.onClickXbox(e);
   }
 
-  onClickAddButton(e: MouseEvent) {
+  onClickXbox(e: MouseEvent) {
+    const { target } = e;
+    if (!(target instanceof HTMLElement)) return;
+    if (target.className === 'card-xbox-img')
+      new ConfirmWindow({
+        onClick: this.onClickConfirmWindowHandler.bind(this),
+        addText: '( 삭제 시 내역 데이터도 삭제됩니다❗️)',
+      });
+  }
+
+  onClickConfirmWindowHandler(e: MouseEvent) {
     const { target } = e;
     if (!(target instanceof HTMLElement)) return;
 
-    if (target.className === 'pay') handleEvent.fire('createhistorymodal'); // 자신의 결제수단 데이터를 넘겨줄것 state는 없어도 됨
+    if (target.className === 'confirm__overlay' || target.className === 'confirm__cancel') {
```

Comment:
includes 를 사용해서 코드를 줄일 수 있습니다 가독성도 더 낫다고 생각합니다.

```suggestion
    if (['confirm__overlay', 'confirm__cancel'].includes(target.className)) {
```

Reply:
오 처음알았는데 좋은 팁이네요!